### ISSUE TITLE
binary value binding fails if data does not contain bytes w/ > 127 value

### ISSUE THREAD

**Issue by zzzeek** [pymssql/pymssql#504]
pymssql 2.1.3
Python 2.7.13 OR Python 3.6.2
SQL Server Linux 2017
Fedora Linux 26
FreeTDS 0.95.81-3-fc26

Below illustrates the pymssql.Binary() wrapper (which appears to just be the bytes type) does not allow some binary values to be inserted without further modifications to the statement:

```
import pymssql

conn = pymssql.connect(
    user="scott",
    password="tiger^5HHH",
    host="mssql2017",
    port=1433
)

cursor = conn.cursor()
cursor.execute("DROP TABLE IF EXISTS t1")
cursor.execute("""
    CREATE TABLE t1 (
        id INTEGER NOT NULL IDENTITY(1,1),
        data VARBINARY(max) NULL,
        PRIMARY KEY (id)
    )
""")

value1 = b'\xf2\x10'
value2 = b'\x10'

# succeeds
cursor.execute(
    "INSERT INTO t1 (data) VALUES (%(value)s)",
    {"value": pymssql.Binary(value1)}
)
print("INSERT #1 passed")


# fails
cursor.execute(
    "INSERT INTO t1 (data) VALUES (%(value)s)",
    {"value": pymssql.Binary(value2)}
)
print("INSERT #2 passed")

```

output:

```
$ python test.py 
INSERT #1 passed
Traceback (most recent call last):
  File "test.py", line 34, in <module>
    {"value": pymssql.Binary(value2)}
  File "pymssql.pyx", line 467, in pymssql.Cursor.execute (pymssql.c:7561)
pymssql.OperationalError: (257, 'Implicit conversion from data type varchar to varbinary(max) is not allowed. Use the CONVERT function to run this query.DB-Lib error message 20018, severity 16:\nGeneral SQL Server error: Check messages from the SQL Server\n')

```

**Comment by AnatoliyAksenov** [comment]
@zzzeek, try to this code to solve your mistake:
```python
cursor.execute(
    "INSERT INTO t1 (data) VALUES (%(value)s)",
    {"value": bytearray(value1)}
)
```

**Comment by zzzeek** [comment]
> @zzzeek, try to this code to solve your mistake:
> 
> ```python
> cursor.execute(
>     "INSERT INTO t1 (data) VALUES (%(value)s)",
>     {"value": bytearray(value1)}
> )
> ```

pymssql should be doing this translation when it is passed a b'' value in python 3.

**Comment by AnatoliyAksenov** [comment]
@zzzeek But it isn't able to do this translation. Only for example:
```python
import os, pymssql

server = os.environ.get("CORP_SERVER")
user = os.environ.get("CORP_USER")
pwd = os.environ.get("CORP_PWD")

conn = pymssql.connect(server=server, user=user, password=pwd, database="corp", charset='CP1251', timeout=100)

cmd = conn.cursor()
with open('test.bin', "wb") as f:
    f.write(b'\00\01\02\03\04\55')
with open('test.text.bin', "wb") as f:
    f.write(b'This is a test binary file contained latin characters')

# create temporary table
cmd.execute('create table ##tbl_test_insert( attachment varbinary(max))')

with open('test.bin', "rb") as f:
    cmd.execute('insert into ##tbl_test_insert (attachment) values(%(attachment)s)', {"attachment": f.read()})

# This entry was ended with an exception: Implicit conversion from data type varchar to varbinary(max) is not allowed
with open('test.text.bin', "rb") as f:
    cmd.execute('insert into ##tbl_test_insert (attachment) values(%(attachment)s)', {"attachment": f.read()})
        
cmd.execute('select attachment from ##tbl_test_insert')
result = cmd.fetchall()
print(result)       
```

**Comment by ryanbentleyxz** [comment]
> import pymssql
> 
> pymssql.Binary = bytearray

Can confirm this fixes my issue as well. This needs to be merged.

**Comment by AbdealiJK** [comment]
> import pymssql; pymssql.Binary = bytearray

This worked well for me too - would be great if it gets merged upstream so I don't need to monkeypatch inside my application

**Comment by termim** [comment]
Fixed in b0988010c3b061bea7e6f14031b3c824145e242c.
